### PR TITLE
reset existing transformations of provided overlap Polygon in Polygon::intersectPolygons

### PR DESCRIPTION
**Pull request by jose-puente**

### CODE REVIEW

**Review by Frosty-J — COMMENTED** [review]
I expect this will be confusing for anyone who hasn't read the #libgdx-contributions channel in Discord, but I have now seen the light and believe it makes sense for transformations to be cleared here, in case someone is reusing `Polygon` objects.

There was a discussion about using `Poolable#reset` left on the table.

File: gdx/test/com/badlogic/gdx/math/IntersectorTest.java [hunk 1]
```diff
@@ -247,4 +246,25 @@ public void testIntersectPolygonsWithVertexLyingOnEdge () {
...
+	@Test
+	public void testIntersectPolygonsWithTransformationsOnProvidedResultPolygon () {
+		Polygon p1 = new Polygon(new float[] {1, -1, 2, -1, 2, -2, 1, -2});
+		Polygon p2 = new Polygon(new float[] {0.5f, -1.5f, 1.5f, -1.5f, 1.5f, -2.5f});
+		Polygon intersectionPolygon = new Polygon(new float[] {0, 0, 0, 0, 0, 0, 0, 0});
+		intersectionPolygon.setScale(5, 5);
+		intersectionPolygon.setOrigin(10, 20);
+		intersectionPolygon.setPosition(-33, -33);
+		intersectionPolygon.setRotation(48);
+		intersectionPolygon.getTransformedVertices();
```
[the remark this answers is not in the document]
Comment (jose-puente) [reply gdx/test/com/badlogic/gdx/math/IntersectorTest.java]: line removed
I guess I forgot to remove it when I was checking if clearing the dirty flag had any impact

**Comment by jose-puente** [comment]
> I expect this will be confusing for anyone who hasn't read the #libgdx-contributions channel in Discord, but I have now seen the light and believe it makes sense for transformations to be cleared here, in case someone is reusing `Polygon` objects.

Any ideas to improve that? Maybe a different name for the test or extra comments in the code?

**Review by obigu — APPROVED** [review]
Good find, thank you. 

As commented previously the PR basically fixes 2 scenarios in which the returned overlap `Polygon` is not correct. 
1. If the provided `overlapPolygon` contained previous transformations those would also be kept on the returned overlap `Polygon` which is not correct since all calculations of the intersection are already made in transformed coordinates. 
2. Even if the `overlapPolygon` had no transformations, if the optimization branch was executed (`System.arraycopy`) the `Polygon` was not set to `dirty` leaving it in an inconsistent state as well.

**Comment by Frosty-J** [comment]
> Any ideas to improve that? Maybe a different name for the test or extra comments in the code?

Providing a description for your pull request would explain the rationale behind it. It's bad practice to leave that field blank for anything more complicated than fixing a typo. Above me is a good example of what it could have said. No real point changing it now, since anyone coming here can scroll down - I'm just conscious of the fact that not everyone lives on Discord.

To be clear, I think this pull request is good to go as-is.

**Pull request merged**